packages/client/mutations/CreateTaskIntegrationMutation.ts

```diff
@@ -187,7 +193,21 @@ const CreateTaskIntegrationMutation: StandardMutation<TCreateTaskIntegrationMuta
         gitlabTaskIntegrationOptimisitcUpdater(store, variables)
       }
     },
-    onCompleted,
+    onCompleted: (data, errors) => {
+      if (onCompleted) {
+        onCompleted(data, errors)
+      }
+      if (data.createTaskIntegration && !data?.createTaskIntegration?.error) {
+        SendClientSegmentEventMutation(atmosphere, 'Task Published', {
+          taskId: data.createTaskIntegration.task?.id,
+          teamId: data.createTaskIntegration.task?.teamId,
+          meetingId: data.createTaskIntegration.task?.meeting?.id,
+          meetingType: data.createTaskIntegration.task?.meeting?.meetingType,
```

Comment:
-1 Ok, now that we established that we don't want wrong information in the 'Task Published' event, let's remove `meetingId` and `meetingType` which come from task. It would be fine if it would come from the current meeting. Following test scenario:
* create task in retrospective
```
analytics.track({
  userId: 'google-oauth2|e3SPYN9W8w',
  event: 'Task Created',
  properties: {
    email: 'georg@parabol.co',
    inMeeting: true,
    meetingId: 'fWD5aT9OHC',
    meetingType: 'retrospective',
    taskId: 'g1PbwncC8E',
    teamId: 'e3SPYTYjg5'
  }
});
```
* publish said task in check-in meeting
```
analytics.track({
  userId: 'google-oauth2|e3SPYN9W8w',
  event: 'Task Published',
  properties: {
    email: 'georg@parabol.co',
    inMeeting: true,
    meetingId: 'fWD5aT9OHC',
    meetingType: 'retrospective',
    service: 'github',
    taskId: 'g1PbwncC8E',
    teamId: 'eOWRlaiWL6'
  }
});
```

Reply:
I'm having some difficulties getting the `meeting` object here. Two ways I can think of:

- Parse the URL to get the meetingId and then grab the meeting object through Relay
- Inject `meetingId` as a variable for CreateTaskIntegration mutation

I like neither. Do you have better way?

Reply:
I don't have a better way. Both are not ideal. I would lean slightly more towards adding `meetingId` to the `createTaskIntegration` mutation to be in line with `createTask` mutation. It feels slightly cleaner.

That being said, there is also
https://github.com/ParabolInc/parabol/blob/47ca01164ad036c5fc1c4773811f82adc094b800/packages/client/utils/meetings/getMeetingPathParams.ts#L25
which might simplify your first option. In the end it is up to you. I also missed a meeting context in the past where I had to do the same path parsing.

Reply:
Alright I chose to use the first option because:
1. it's less code (if I add `meetingId` to `CreateTaskIntegration` mutation then I need to add `meetingId` to all the places where we call `CreateTaskIntegration`)
2. even if we add `meetingId` to `CreateTaskIntegration`, from client side we still need to get the `meetingId` from somewhere. Since `CreateTaskIntegrationMutation` is fired through `TaskFooterIntegrateMenu`, which could be placed everywhere. So we are back to parsing `meetingId` from URL.